librepo/yum.c

```diff
@@ -131,8 +131,18 @@ lr_yum_repo_update(LrYumRepo *repo, const char *type, const char *path)
 
 /* main bussines logic */
 
+gint
+compare_records(gconstpointer a, gconstpointer b)
+{
+    GSList* elem = (GSList*) a;
+    LrYumRepoMdRecord* yum_record = (LrYumRepoMdRecord*) elem;
```

Comment:
Please access explicitly the ``data`` element of GSList here:

    elem->data;

See: https://developer.gnome.org/glib/stable/glib-Singly-Linked-Lists.html#GSList

You can directly merge lines 137-138 by:

    LrYumRepoMdRecord* yum_record = (GSList*) a->data;